src/commands/app/upgrade.tsx

```diff
@@ -0,0 +1,170 @@
+import { ExecRenderBaseCommand } from "../../rendering/react/ExecRenderBaseCommand.js";
+import {
+  appInstallationArgs,
+  withAppInstallationId,
+} from "../../lib/app/flags.js";
+import { projectFlags } from "../../lib/project/flags.js";
+import { Flags, ux } from "@oclif/core";
+import React, { ReactNode } from "react";
+import { Text } from "ink";
+import { getAppFromUuid, getAppVersionFromUuid } from "../../lib/app/uuid.js";
+import {
+  getAvailableTargetAppVersionFromExternalVersion,
+  getLatestAvailableTargetAppVersionForAppVersionUpgradeCandidates,
+} from "../../lib/app/versions.js";
+import {
+  makeProcessRenderer,
+  ProcessFlags,
+  processFlags,
+} from "../../rendering/process/process_flags.js";
+import { Success } from "../../rendering/react/components/Success.js";
+
+export class UpgradeApp extends ExecRenderBaseCommand<typeof UpgradeApp, void> {
+  static description = "Upgrade target app installation to target version";
+  static args = {
+    ...appInstallationArgs,
+  };
+  static flags = {
+    "target-version": Flags.string({
+      description: "target version to upgrade target app to",
+      required: true,
+      default: "latest",
+    }),
+    "show-candidates": Flags.boolean({
+      description: "list version candidates the given app can be upgraded to",
+    }),
+    force: Flags.boolean({
+      char: "f",
+      description: "Do not ask for confirmation",
+    }),
+    ...projectFlags,
+    ...processFlags,
+  };
+
+  protected async exec(): Promise<void> {
+    const process = makeProcessRenderer(
+      this.flags as ProcessFlags,
+      "App Upgrade",
+    );
+    const appInstallationId = await withAppInstallationId(
+      this.apiClient,
+      UpgradeApp,
+      this.flags,
+      this.args,
+      this.config,
+    );
+
+    const currentAppInstallationResponse =
+      await this.apiClient.app.getAppinstallation({
+        appInstallationId: appInstallationId,
+      });
```

Comment:
`assertStatus` to assert the expected return status code; this should also make the subsequent type casts unnecessary, because the compiler should be able to figure out the data types on its own.

Reply:
I'll have to keep an eye out for correct assertion, and will do that.
For this explicit instance the whole api call was put into a wrapper function with correct assertion in 2fe9d63108be465704faa55aae270c26dfb4f3ce